src/Autocomplete.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component, PropTypes } from 'react';
+import cx from 'classnames';
+import Icon from './Icon';
+import Row from './Row';
+import Col from './Col';
+
+class Autocomplete extends Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      value: ''
+    };
+
+    this.renderIcon = this.renderIcon.bind(this);
+    this.renderDropdown = this.renderDropdown.bind(this);
+    this._onChange = this._onChange.bind(this);
+  }
+
+  renderIcon (icon, iconClassName) {
+    return <Icon className={iconClassName}>{icon}</Icon>;
+  }
+
+  renderDropdown (data) {
+    const { value } = this.state;
+
+    let li = null;
+    if (value.length) {
+      li = Object.keys(data).map((key, idx) => {
+        const index = key.toUpperCase().indexOf(value.toUpperCase());
+        if (index !== -1 && value.length < key.length) {
+          return (
+            <li key={key + '_' + idx}>
+              {data[key] ? <img src={data[key]} className="right circle"/> : null}
+              <span>
+                {index !== 0 ? key.substring(0, index) : ''}
+                <span className='highlight'>{value}</span>
+                {key.length !== index + value.length ? key.substring(index + value.length) : ''}
+              </span>
+            </li>
+          );
+        }
+      })
+    }
+
+    return li;
+  }
+
+  _onChange (evt) {
+    this.setState({ value: evt.target.value });
+  }
+
+
+  render () {
+    const {
+      children,
+      className,
+      title,
+      data,
+      icon,
+      iconClassName,
+      s,
+      m,
+      l,
+      offset,
+      ...props
+    } = this.props;
+
+    const _id = 'autocomplete-input';
+    const _autocomplete = 'autocomplete';
+    const _ul = 'autocomplete-content dropdown-content';
+
+    return (
+      <Row>
+        <Col s={s} m={m} l={l} offset={offset} className={cx('input-field', className)} {...props}>
+          {icon && this.renderIcon(icon, iconClassName)}
+          <input type="text" id={_id} className={_autocomplete} onChange={this._onChange}/>
+          <label htmlFor={_id}>{title}</label>
+          <ul className={_ul}>{this.renderDropdown(data)}</ul>
+          {children}
+        </Col>
+      </Row>
+    );
+  }
+}
+
+Autocomplete.propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
+  /*
+   * The title of the autocomplete component.
+   */
+  title: PropTypes.string,
+  /*
+   * An object with the keys of the items to match in autocomplete
+   * The values are either null or a location to an image
+   */
+  data: PropTypes.object.required,
+  /*
+   * Optional materialize icon to add to the autocomplete bar
+   */
+  icon: PropTypes.string,
+  iconClassName: PropTypes.string,
+  s: PropTypes.number,
+  m: PropTypes.number,
+  l: PropTypes.number,
+  offset: PropTypes.string
+};
+
+Autocomplete.defaultProps = {
+  s: 12,
+  m: 12,
+  l: 12
+};
+
+export default Autocomplete;
```

Comment:
This will suppress Github's warnings about no new line at end of file http://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file